### PR TITLE
Create struct service.Configuration

### PR DESCRIPTION
**Pull request by NicolasMahe**
Closes https://github.com/mesg-foundation/engine/issues/1104

This PR add a new struct `service.Configuration` that is also used by `service.Dependency` by inheritance and update the codebase to propagate the modif.

### CODE REVIEW

File: service/type.go [hunk 1]
```diff
@@ -76,32 +76,38 @@ type Service struct {
...
+// Dependency represents a Docker container and it holds instructions about
+// how it should run.
+type Dependency struct {
+	*Configuration
+
+	// Key is the key of dependency.
+	Key string `hash:"name:d1" validate:"printascii"`
+
+	// Image is the Docker image.
+	Image string `hash:"name:d2" validate:"printascii"`
```
Comment (NicolasMahe) [inline service/type.go]: As it is using inheritance, I choose to name the hash with the prefix "d" so it doesn't override future properties added to Configuration struct.

Comment (krhubert) [reply service/type.go]: I've just tested structhash package and we don't need to prefix for hash. With struct composition when the two fields with the same name hash is different.
```go
package main

import (
    "fmt"

    "github.com/cnf/structhash"
)

type A struct {
    N string `hash:"name:1"`
    M string `hash:"name:2"`
    B
}

type B struct {
    M string `hash:"name:2"`
}

func main() {
    fmt.Printf("%x\n", structhash.Dump(A{N: "n", M: "m", B: B{M: "m"}}, 0))
    fmt.Printf("%x\n", structhash.Dump(A{N: "n", M: "m-", B: B{M: "m"}}, 0))

    fmt.Printf("%x\n", structhash.Sha1(A{N: "n", M: "m", B: B{M: "m"}}, 0))
    fmt.Printf("%x\n", structhash.Sha1(A{N: "n", M: "m-", B: B{M: "m"}}, 0))
}

```
➜  go run main.go
7b313a226e222c323a226d222c423a7b323a226d227d7d
7b313a226e222c323a226d2d222c423a7b323a226d227d7d
338eec5b413091c4db0e9aa057e861549ca45d9e
11ac72d3248a5dd00d65a1b1640a1fb6bc8326c3
```

File: service/type.go [hunk 2]
```diff
@@ -76,32 +76,38 @@ type Service struct {
...
+	Env []string `hash:"name:6" validate:"unique,dive,env"`
+}
+
+// Dependency represents a Docker container and it holds instructions about
+// how it should run.
+type Dependency struct {
+	*Configuration
+
+	// Key is the key of dependency.
+	Key string `hash:"name:d1" validate:"printascii"`
```
Comment (NicolasMahe) [inline service/type.go]:
```suggestion
	Key string `hash:"name:1" validate:"printascii"`
```

File: service/type.go (hunk 1 above)
Comment (NicolasMahe) [inline service/type.go]:
```suggestion
	Image string `hash:"name:2" validate:"printascii"`
```

Comment (NicolasMahe) [reply service/type.go]: nice catch 👍  I removed the prefix

**Pull request merged**